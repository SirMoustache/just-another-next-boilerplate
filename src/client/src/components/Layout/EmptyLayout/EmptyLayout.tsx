/**
 * Absolute imports
 */
import React from 'react';
import styled from 'styled-components';
import { motion } from 'framer-motion';

export type BaseProps = {
  justify?: 'center' | 'flex-end' | 'flex-start';
  align?: 'center' | 'flex-end' | 'flex-start';
};

const Base = styled(motion.main)<BaseProps>`
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: ${({ justify = 'center' }) => justify};
  align-items: ${({ align = 'center' }) => align};
`;

export type EmptyLayoutProps = {
  children?: React.ReactNode;
};

const EmptyLayout = ({ children }: EmptyLayoutProps) => {
  return (
    <Base initial="initial" animate="animate" exit={{ opacity: 0 }}>
      {children}
    </Base>
  );
};

export default EmptyLayout;
