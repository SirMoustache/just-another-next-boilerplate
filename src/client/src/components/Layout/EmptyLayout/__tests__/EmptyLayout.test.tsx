/**
 * Absolute imports
 */
import React from 'react';
import { render, cleanup } from '@testing-library/react';

/**
 *
 */
import EmptyLayout from '../EmptyLayout';

const children = <span>Test</span>;
const renderComponent = (props = {}) =>
  render(<EmptyLayout {...props}>{children}</EmptyLayout>);

describe('<EmptyLayout />', () => {
  afterEach(cleanup);

  it('should match snapshot', () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  it('should render a <main> tag', () => {
    const { container } = renderComponent();
    expect(container.querySelector('main')).not.toBeNull();
  });

  it('should have render children', () => {
    const { container } = renderComponent();
    // @ts-ignore
    expect(container.querySelector('main').children).toHaveLength(1);
  });
});
