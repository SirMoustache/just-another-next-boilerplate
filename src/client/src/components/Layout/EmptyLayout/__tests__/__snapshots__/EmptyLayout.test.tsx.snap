// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyLayout /> should match snapshot 1`] = `
<div>
  <main
    class="sc-AxjAm eIoqnP"
  >
    <span>
      Test
    </span>
  </main>
</div>
`;
